src/main/utility/childpid_watcher.rs

```diff
@@ -249,11 +242,15 @@ impl ChildPidWatcher {
     /// an unrelated process with a recycled `pid`.
     pub fn register_pid(&self, pid: Pid) {
         let mut inner = self.inner.lock().unwrap();
-        let pidfd =
-            pidfd_open(pid).unwrap_or_else(|e| panic!("pidfd_open failed for {pid:?}: {e:?}"));
-
-        let event = EpollEvent::new(EpollFlags::EPOLLIN, pid.as_raw().try_into().unwrap());
-        self.epoll.add(&pidfd, event).unwrap();
+        let pidfd = rustix::process::pidfd_open(pid.into(), PidfdFlags::empty())
+            .unwrap_or_else(|e| panic!("pidfd_open failed for {pid:?}: {e:?}"));
```

Comment:
I think it would be good to make the pidfd non-blocking. I have had bad experiences from epoll saying something is ready to read/write, and then the read/write syscall blocks. By adding it for the pidfd, we should catch any errors when the eventual waitid is run (if it were to block with a blocking pidfd, we'd never know there was an error since the thread would just stall, but if it's a non-blocking pidfd and waitid returns `EAGAIN` instead, we know there's a bug).

Reply:
Seems like a good idea; done